src/lifters/wideningTokenLifter.ml

```diff
@@ -12,13 +12,13 @@ module Token = WideningToken
 module TS = SetDomain.ToppedSet (Token) (struct let topname = "Top" end)
 
 (** Reference to current {!add} implementation. Maintained by {!Lifter}. *)
-let add_ref: (Token.t -> unit) ref = ref (fun _ ->
+let add_ref: (Token.t -> unit) Domain.DLS.key = Domain.DLS.new_key (fun () _ ->
```

Comment:
Similar question here: we have a lot of top-level `ref`s around, but only this one is made domain-local. Are all others fine then?